### PR TITLE
Expanding HTTP section.

### PR DESCRIPTION
**Pull request by yuri**

### CODE REVIEW

**Comment by yuri** [comment]
@SethDavenport Please review after the fact.

File: handout/05-rest-apis.md [hunk 1]
```diff
@@ -1,33 +1,189 @@
...
-we will start writing custom services using Angular’s `.factory()` method.
-Server interaction introduces *asynchronicity*, so we look at *promises*,
-which is currently the preferred approach to handling asynchronous code in
-JavaScript. We will look at *chaining* promises, a well as error handling. In
-addition to reusing promises created by `$http.get`, we will also create our
-own using `$q`. We will also consider use of promises in `$timeout` and
-`$interval`.
+Most AngularJS apps will need to interact with a backend system to achieve
+their objectives. By far the most common protocol for this is HTTP (or,
+preferrably, it's secure version, HTTPS). While this is the same protocol that
```
Comment (SethDavenport) [inline handout/05-rest-apis.md]: preferably, its

File: handout/05-rest-apis.md [hunk 2]
```diff
@@ -1,33 +1,189 @@
...
+was employed by the previous generation of web applications, it tends to be
+used rather differently in a context of a modern single-page application
+framework such as AngularJS.
 
-## Talking to a Rest Server via Postman
+## Talking to a RESTful Server via Postman
 
-Before we begin talking to the server, let's get comfortable interacting with
-the REST API using Postman.
+In most cases, the Angular application and the server would not be exchanging
```
Comment (SethDavenport) [inline handout/05-rest-apis.md]: will not be

File: handout/05-rest-apis.md [hunk 3]
```diff
@@ -1,33 +1,189 @@
...
+to represent a data "resource" in some collection of resources. The
+application can use HTTP to create a resource, retrieve an existing one,
+update it, or delete it.
 
-If you do not have Postman install, get it here:
+Before we begin talking to the server from our web application, let's get
+comfortable interacting with the REST API using Postman.
 
-    <http://http://www.getpostman.com/>
+If you do not have Postman install, get it here: <http://http://www.getpostman.com/>
```
Comment (SethDavenport) [inline handout/05-rest-apis.md]: install[ed]

File: handout/05-rest-apis.md [hunk 4]
```diff
@@ -1,33 +1,189 @@
...
+An HTTP client interacts with server resources using HTTP methods:
 
-## $http.get
+__GET__ to retrieve some resource (or resources). E.g.:
+
+```
+  GET http://ngcourse.herokuapp.com/api/v1/tasks/5491e9da995e33455a7307e2
+```
+
+The server will respond with the a set of object represenging matching resources:
```
Comment (SethDavenport) [inline handout/05-rest-apis.md]: object[s]

s/represenging/representing/

File: handout/05-rest-apis.md [hunk 5]
```diff
@@ -1,33 +1,189 @@
...
+```
+  Content-Type: text/plain
+```
+
+The good news, however, is that Angular will assume that you are using JSON
+and will handle the details for you.
+
+## Status Codes
+
+An HTTP response must contain a numberic status code that indicate whether the
```
Comment (SethDavenport) [inline handout/05-rest-apis.md]: s/numberic/numeric/

File: handout/05-rest-apis.md [hunk 6]
```diff
@@ -1,33 +1,189 @@
...
+```
+
+Or on Window:
+
+```
+  chrome.exe --disable-web-security
+```
+
+Same-origin policy does _not_ apply to apps built with Cordova. However, you
+will run into this option when testing those apps in a web browser.
```
Comment (SethDavenport) [inline handout/05-rest-apis.md]: Interestingly, this is only a problem in chrome.  You can test the cordova app in firefox served up from file:/// just fine.

Not relevant to this course, but I just thought I'd mention it.

**Comment by SethDavenport** [comment]
Bunch of typos, but the content is much better.  LGTM.

**Pull request merged**